parquet-column/src/main/java/parquet/filter2/recordlevel/IncrementallyUpdatedFilterPredicate.java

```diff
@@ -61,6 +64,24 @@
 
     private boolean result = false;
     private boolean isKnown = false;
+    private BitSet dictionaryResults;
+
+    /** Populates array of BitSet based on whether values pass filter */
+    void setDictionary(Dictionary dictionary) {
+      dictionaryResults = new BitSet(dictionary.getMaxId()+1);
+      for (int i = 0, end = dictionary.getMaxId()+1; i < end; i++) {
+        dictionaryResults.set(i, update(dictionary, i));
+      }
+    }
+
+    /** Update based on element from dictionary */
+    void updateFromDictionary(int dictionaryId) {
+      setResult(dictionaryResults.get(dictionaryId));
+    }
+
+    /** Does the element in the dictionary pass the filter */
+    abstract protected boolean update(Dictionary dictionary, int dictionaryId);
```

Comment:
this is strange to me. I think the goal it to make things more efficient by know the dictionary in advance.